pyomo/contrib/parmest/utils/model_utils.py

```diff
@@ -201,3 +201,46 @@ def convert_params_to_vars(model, param_names=None, fix_vars=False):
     # solver.solve(model)
 
     return model
+
+
+def update_model_from_suffix(suffix_obj: pyo.Suffix, values):
+    """
+    Overwrite each variable/parameter referenced by ``suffix_obj`` with the
+    corresponding value in ``values``. The provided values are expected to
+    be in the same order as the components in the suffix from when it was
+    created.
+
+    Parameters
+    ----------
+    suffix_obj : pyomo.core.base.suffix.Suffix
+        The suffix whose *keys* are the components you want to update.
+        Call like ``update_from_suffix(model.unknown_parameters, vals)``.
+    values : iterable of numbers
+        New numerical values for the components referenced by the suffix.
+        Must be the same length as ``suffix_obj``.
+
+    Notes
+    -----
+    Measurement error is a special case: instead of updating the value of the
+    keys (variables/parameters), it updates the value stored in the suffix itself.
```

Comment:
```suggestion
    The measurement_error suffix is a special case: instead of updating the value of the
    keys (variables/parameters), it updates the value stored in the suffix itself.
```